cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORGenerator.java

```diff
@@ -501,42 +536,6 @@ public final void writeFieldId(long id) throws IOException {
         _writeLongNoCheck(id);
     }
 
-    /*
-    /**********************************************************
-    /* Overridden methods, copying with tag-awareness
-    /**********************************************************
-     */
-
-    /**
-     * Specialize {@link JsonGenerator#copyCurrentEvent} to handle tags.
-     */
-    @Override
-    public void copyCurrentEvent(JsonParser p) throws IOException {
```

Comment:
Where are these implemented now? Without overrides tag information will be lost if buffering (although TBH this won't work with general `TokenBuffer` anyway)

Reply:
This is explained in the message for the first commit in the series:

> Copying tags can be dangerous in two main cases:
> 1. Jackson creates its own tags, such as for BigNum or the self describe
   tag.
> 2. The semantic meaning of an element changes. For example, converting
   between stringrefs and no stringrefs.
>
> In some cases it would result in duplicate tags, while in others it may
cause completely incorrect tags, such as a stringref tag applying to a
string rather than an integer. Before nested tag support was added this
could generate files that Jackson could not parse, though even with nested
tag support it may cause undefined behavior.

Reply:
Ok, not sure about part "Jackson creates its own tags" here (they come from CBOR spec)?

But aside from that, `copyCurrentEvent` is used to copy exact contents of input. I guess it can be problematic for shared String references as unlike other tags that only related to specific token/value without context/state, string-refs do have context (dictionary being collected).

Given that no unit test fails I guess this is fine. And if this becomes a problem can re-implement methods as necessary.

Reply:
I mean standard CBOR tags that are created based on the generation code within the `CBORGenerator` class.

This behavior is because the event forwarding still goes through the standard generator interface. For example, let's say you are reading a value that gets interpreted as a BigNum. A CBOR tag is created to decorate the binary string value that holds the data to denote that it represents a BigNum. This logic will be executed regardless of whether an existing BigNum tag exists or not, resulting in *two* BigNum tags. (one from the tag copy, one created from `CBORGenerator`)

As more tags are supported by Jackson, the more dangerous that behavior can get. For example, let's say you're converting from a stringref encoded CBOR to one without stringrefs. It would read an integer index corresponding to a string and convert it to a "real" string, but the tag copying logic it would *also* apply the tag denoting that the value is a stringref index. This is invalid as the stringref tag is intended to apply to an integer, while it would apply to a string instead.

Reply:
Ah. Ok, yes.